harness/determined/common/context.py

```diff
@@ -126,13 +128,18 @@ def add_item(f: bindings.v1File) -> None:
         for file in files:
             file_path = pathlib.Path(parent).joinpath(file)
             file_rel_path = file_path.relative_to(root_path)
+            parent_rel_path = pathlib.Path(parent).relative_to(root_path)
 
-            # If the file is the .detignore file or matches one of the
-            # paths specified in .detignore, then ignore it.
+            # If the file is the .detignore file, it matches one of the
+            #  paths specified in .detignore, or if its parent directory does, then ignore it.
```

Comment:
hey, that's a tricky bug you are detecting here.  Nice catch.

Though I don't think you're fully solving it.  If `/bad/` is ignored, the `/bad/a/b/c/d` should also be ignored, though checking the direct parent isn't going to be enough.

Really what you need to do is avoid entering an ignored directory in the first place.

According to `os.walk()`'s [docs](https://docs.python.org/3/library/os.html#os.walk), that is possible by removing `directory` from `dirs:

> When topdown is True, the caller can modify the dirnames list in-place (perhaps using [del](https://docs.python.org/3/reference/simple_stmts.html#del) or slice assignment), and [walk()](https://docs.python.org/3/library/os.html#os.walk) will only recurse into the subdirectories whose names remain in dirnames; this can be used to prune the search

Reply:
Woah, I did not know about that feature of `walk()`. Pretty slick!

Reply:
cool, now let's undo changes to this comment, since you are no longer changing the code under the comment

Reply:
Oops! Good catch.